src/main/groovy/com/avast/gradle/dockercompose/ComposeSettings.groovy

```diff
@@ -117,6 +119,16 @@ class ComposeSettings {
     }
 
     void isRequiredBy(Task task) {
+        if (inheritInputAndOutputs) {
+            upTask.inputs.files task.inputs.files
+            downTask.inputs.files task.inputs.files
```

Comment:
Also please add dependency to `docker-compose.yml` file(s) - check the two methods called [here](https://github.com/avast/gradle-docker-compose-plugin/blob/master/src/main/groovy/com/avast/gradle/dockercompose/ComposeExecutor.groovy#L96).